kafka-cron/README.md

```diff
@@ -0,0 +1,227 @@
+
+Run some instances of your consumer program that read from your retry queues (we can make this a command-line option in your consumer).
+
+Define a job that fails and observe your retry consumers retrying and eventually discarding it.
+
+Define a job that randomly fails some percent of the time, and observe your retry consumers retrying and eventually completing it.
+
+### Part 4: Monitoring and Alerting
+
+In software operations, we want to know what our software is doing and how it is performing.
+
+One very useful technique is to have our program export metrics. Metrics are basically values that our program makes available (the industry standard is to export and scrape over HTTP).
+
+Specialised programs, such as Prometheus, can then fetch metrics regularly from all the running instances of our program, store the history of these metrics, and do useful arithmetic on them (like computing rates, averages, and maximums). We can use this data to do troubleshooting and to alert if things go wrong.
+
+Read the [Overview of Prometheus](https://prometheus.io/docs/introduction/overview/) if you are not familiar with Prometheus.
+
+The [Prometheus Guide to Instrumenting a Go Application](https://prometheus.io/docs/guides/go-application/) describes how to add metrics to a Golang application.
+
+First, consider:
+ * What kinds of things may go wrong with our system? (it is useful to look at errors your code is handling, as inspiration)
+ * What would users' expectations be of this system?
+ * What metrics can we add that will tell us when the system is not working as intended?
+ * What metrics can we add that might help us to troubleshoot the system and understand how it is operating? Read back through the first three parts of this exercise to try and identify the properties of the system that we might want to know about.
+
+Asking these questions should guide us in designing the metrics that our consumers and producer should export.
+Think about what kinds of problems can happen both in the infrastructure - Kafka, your consumers and producers - and in the submitted jobs. 
+
+Add metrics to your programs. Verify that they work as expected using `curl` or your web browser.
+
+#### Running the Prometheus JMX Exporter to get Kafka metrics
+Kafka doesn't export Prometheus metrics natively. However, we can use the official 
+[Prometheus JMX exporter](https://github.com/prometheus/jmx_exporter) to expose its metrics.
+
+The Prometheus JMX exporter can run as a Java agent (alongside a Java program such as Kafka) or else as a standalone HTTP server, which collects metrics from a JVM running elsewhere and re-exports them as Prometheus metrics. If you're using [conduktor/kafka-stack-docker-compose](https://github.com/conduktor/kafka-stack-docker-compose) as suggested above then your image contains the `jmx_prometheus_javaagent` already. 
```

Comment:
I'm not sure we've introduced the concept of a JVM anywhere before - may be worth a one-sentence "Kafka is a Java program, Java programs run in a host process called a JVM." ?